### PR TITLE
WT-1957 Ignore casing on address check when finding balance in smart checkout

### PR DESCRIPTION
**Pull request by imx-mikhala**
# Summary

Lower cases the addresses when doing comparison check for finding the users balance of a given token item requirement

### CODE REVIEW

**Review by sharonsheah — APPROVED** [review]

**Review by dreamoftrees — DISMISSED** [review]
@imx-mikhala nice one, the fix looks good. Just curious what the scenario would be where this could happen? I worry if we need to compare addresses like this there are a few different places where comparison happens and we might get varying results?

File: packages/checkout/sdk/src/smartCheckout/routing/indexer/fetchL1Representation.ts [hunk 1]
```diff
@@ -46,9 +47,8 @@ export const fetchL1Representation = async (
     contractAddress: l2address,
   });
 
-  // TODO: When bridge is ready we need to understand how L2 ETH will be mapped back to L1 ETH
```
Comment (imx-mikhala) [inline packages/checkout/sdk/src/smartCheckout/routing/indexer/fetchL1Representation.ts:49]: Is this still something we need to look into @dreamoftrees ?

Comment (dreamoftrees) [reply packages/checkout/sdk/src/smartCheckout/routing/indexer/fetchL1Representation.ts:49]: All good, I think this is sorted now.

**Review by dreamoftrees — APPROVED** [review]
Nice one @imx-mikhala 👍

**Review by dreamoftrees — DISMISSED** [review]
👍

**Review by dreamoftrees — APPROVED** [review]
👍

**Review by sharonsheah — APPROVED** [review]

**Pull request merged**